### ISSUE TITLE
Adjustable gear cause constant FAR revoxelization in flight

### ISSUE THREAD

**Issue by seanyoung247** [shadowmage45/KerbalFoundries2#12]
The adjustable gear, when added to a craft, cause FAR to constantly update the crafts voxels. Removing just the gear fixes the issue.

**Comment by shadowmage45** [comment]
Thanks for the report.  Seems likely that it is caused by some incorrect code in the animation/drag updating routines.  Will do some investigation and let you know how it turns out.

**Comment by shadowmage45** [comment]
Fixed in dev code, will be available with the next KSPWheels release.

**Comment by seanyoung247** [comment]
Four minutes! It took you four minutes to fix! I hate it when mod developers just can't be bothered and leave you hanging :p

Seriously great thank you.

**Comment by Toonu** [comment]
Hello, this issue occured to me again in 1.7.3. There is periodic lag when using KF and FAR together and it works fine when without it.


[KSP.log](https://github.com/shadowmage45/KerbalFoundries2/files/4173120/KSP.log)

Without wheels
https://youtu.be/HEYlj63l3Q0

With wheels
https://youtu.be/8inkJqLAcj0

**Comment by Toonu** [comment]
@shadowmage45 forgot to ping you, not sure if you notice closed request comments without that? Sorry if that is the case!

**Comment by shadowmage45** [comment]
With something like this, it is best to open a new issue.  The original one was obviously resolved, and whatever is occurring likely isn't simply the 'same problem', but a new problem.

I'm going to go-ahead and re-open this one for now, because you already posted your logs/etc; but in the future, if an issue is more than ~month being closed (let alone years), simply open a new one :)

A good question is -- is there an official FAR release for KSP 1.7 / 1.8?  Last I knew, there hadn't been one in forever.

I should also add that I do not use or officially support FAR with any of my mods, so any solution is very low priority for me personally.  If you had information on where the issue might be in the code, feel free to add that in, might go a long way towards getting a solution implemented.

**Comment by Toonu** [comment]
Yes I am using official fork for 1.7.2 and there is already FAR for 1.8 as well. Now even pre-release for 1.9 when I checked their git releases. I am using v0.15.11.2 
https://github.com/dkavolis/Ferram-Aerospace-Research/releases

Sorry for misplaced reopened issue, yeah next time sir... 

This issue was originally different I guess and was fixed, so I hope this issue will be resolved aswell even when you do not support FAR explicitly. Thank you.

**Comment by shadowmage45** [comment]
Strange, I'm only seeing the geometry rebuild function called in one place, and from reading through the code, it should only be called during actual animation changes:
https://github.com/shadowmage45/KSPWheel/blob/master/VSProject/KSPWheel/PartModules/KSPWheelAdjustableGear.cs#L465-L469

**Comment by Toonu** [comment]
So is there any way to help you fix this issue? I know it is not priority, but if I could provide any more info or way to replicate it?

**Comment by shadowmage45** [comment]
Basically, it just comes down to me finding the time (and patience) to download and setup FAR, turn on some debugging/add some debug statements, and figure out why the geometry update function is being called when it shouldn't be.  In the past, every time I've dealt with FAR, it takes hours to setup and get working (due to its list of dependencies), and even simple debugging turns into a multi-day-long painful experience.

I would do this _now_ if I didn't have to deal with FAR and could test/fix the issue in a stock environment.  Hmm... perhaps I can?  (I don't actually need FAR installed to see if/when the function is being called)...

**Comment by shadowmage45** [comment]
I'm not seeing any issues during testing.  KSPWheel is only calling the FAR geometry rebuild method while it is actively animating, and even then it doesn't even call it every frame (more like every 5-10 frames):

![image](https://user-images.githubusercontent.com/1503399/75382367-9a48f500-5897-11ea-910c-36008fcee821.png)

**Comment by shadowmage45** [comment]
@ferram4
Any insight into what might have changed in FAR that could be causing this?  The KSPWheel code for these updates is exactly the same as when this issue was first reported and fixed; nothing has changed on this end that I can see.

**Comment by Toonu** [comment]
@shadowmage45 have you tried it in flight? Because actually it doesn't lag that much (or at least less) while landed or on ground, it is maybe affected by speed of the craft also? I can maybe take another video of the issue...

**Comment by shadowmage45** [comment]
KSPWheel doesn't do anything differently based on craft speed or landed status.

So if you are seeing differences in those instances, it would point towards the issue being on FAR's side of things.

----

Who is actually managing the FAR codebase these days?  Looks like Ferram hasn't done a commit in >2 years, so not sure who I should tag for more info....

At this point, the problem really should be reported in the FAR thread; and when the FAR author knows what is going wrong, they can contact me with the details and we'll work through it from there.

But until I have more information, I'm not going to spend time downloading and dealing with FAR or doing any debugging myself.  As far as I know it is not my problem; I hate to take that stance, but I also hate dealing with external mods that I don't use or support causing me to spend time that I could put to more productive use elsewhere.

**Comment by Toonu** [comment]
dkavolis is developing it in his repo https://github.com/dkavolis/Ferram-Aerospace-Research, I will submit bug report there. Thanks for help sir.

**Comment by shadowmage45** [comment]
Good to know, and thanks.  I have no problem fixing anything that might need to be on my end, but from looking at the code alone, I have no idea what that might be (as it seems to work fine in isolation).
